$OpenBSD: patch-src_events_c,v 1.1.2.1 2009/08/13 01:27:27 william Exp $

Security fix for CVE-2008-4776:  correctly check the length of contact names

--- src/events.c.orig	Tue Jun 17 16:29:54 2008
+++ src/events.c	Fri Jul 10 20:03:03 2009
@@ -621,7 +621,7 @@ static int gg_watch_fd_connected(struct gg_session *se
 				if (GG_S_D(n->status)) {
 					unsigned char descr_len = *((char*) n + sizeof(struct gg_notify_reply77));
 
-					if (descr_len < length) {
+					if (sizeof(struct gg_notify_reply77) + descr_len <= length) {
 						if (!(e->event.notify60[i].descr = malloc(descr_len + 1))) {
 							gg_debug_session(sess, GG_DEBUG_MISC, "// gg_watch_fd_connected() not enough memory for notify data\n");
 							goto fail;
@@ -744,7 +744,7 @@ static int gg_watch_fd_connected(struct gg_session *se
 				if (GG_S_D(n->status)) {
 					unsigned char descr_len = *((char*) n + sizeof(struct gg_notify_reply60));
 
-					if (descr_len < length) {
+					if (sizeof(struct gg_notify_reply60) + descr_len <= length) {
 						if (!(e->event.notify60[i].descr = malloc(descr_len + 1))) {
 							gg_debug_session(sess, GG_DEBUG_MISC, "// gg_watch_fd_connected() not enough memory for notify data\n");
 							goto fail;
